.github/workflows/test.yml

```diff
@@ -26,16 +26,59 @@ jobs:
         shell: bash
         run: make lint
 
+  prepare-unit-test-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-unit-test-matrix-push-pr.outputs.matrix || steps.set-unit-test-matrix-workflow-dispatch.outputs.matrix }}
+    steps:
+      - name: Set unit test matrix for push or pull_request
+        if: github.event_name == 'push' || github.event_name == 'pull_request'
+        id: set-unit-test-matrix-push-pr
+        run: |
+          echo "::set-output name=matrix::{ \
+                  \"python-version\": [\"3.8\", \"3.9\", \"3.10\", \"3.11\"], \
+                  \"os\": [\"ubuntu-latest\", \"macos-latest\"], \
```

Comment:
I guess this workaround is needed and this can't be more readable?

Could we have 2 separate files for this and import common tasks? I recall Chirag did it like this?

Reply:
I created a separate script for the creation of these matrices and imported in the unit and integration test matrix when needed Hopefully this will be much more readable and easy to adjust if needed in the future.